### PR TITLE
feat: Route Coordination (RFC 0091) - Base service structure implementation

### PR DESCRIPTION
**Pull request by rolsonquadras**
- Create New Service for Route Coordination Spec
- Implement Accept/Name functions

part of #837 

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

### CODE REVIEW

File: pkg/didcomm/protocol/route/service.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+// constants for route coordination spec types
+const (
+	// Coordination route coordination protocol
+	Coordination = "routecoordination"
+
+	// RouteCoordinationSpec defines the route coordination spec
+	CoordinationSpec = "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/routecoordination/1.0/"
```
Comment (troyronda) [inline pkg/didcomm/protocol/route/service.go]: Suggest just using `https://didcomm.org`

Comment (rolsonquadras) [reply pkg/didcomm/protocol/route/service.go]: [Updated](https://github.com/hyperledger/aries-framework-go/compare/d9207c876b65c0cae56d2aae1a2efbddcf8f0996..d261ea92a76770ea17bc427f183b9cbda779bb62#diff-7cc3df019152fe2c557826a312cc0a12R25). Note - [RFC](https://github.com/hyperledger/aries-rfcs/tree/master/features/0211-route-coordination#route-request) still used DID.

File: pkg/didcomm/protocol/route/support_test.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+func (p *mockProvider) StorageProvider() storage.Provider {
+	if p.openStoreErr != nil {
+		return &mockstore.MockStoreProvider{ErrOpenStoreHandle: p.openStoreErr}
+	}
+
+	return mockstore.NewMockStoreProvider()
+}
+
+func (p *mockProvider) InboundTransportEndpoint() string {
+	return "ws//example.com"
```
Comment (troyronda) [inline pkg/didcomm/protocol/route/support_test.go]: is this missing a :?

Comment (rolsonquadras) [reply pkg/didcomm/protocol/route/support_test.go]: [fixed](https://github.com/hyperledger/aries-framework-go/compare/d261ea92a76770ea17bc427f183b9cbda779bb62..f71d3c668644822b868c0ebec4387a83ea9d3b5d#diff-6f60e8f13e1dea1aa90429112324187fR36).

File: pkg/didcomm/protocol/route/support_test.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	mockdispatcher "github.com/hyperledger/aries-framework-go/pkg/internal/mock/didcomm/dispatcher"
+	mockkms "github.com/hyperledger/aries-framework-go/pkg/internal/mock/kms"
+	mockstore "github.com/hyperledger/aries-framework-go/pkg/internal/mock/storage"
+	"github.com/hyperledger/aries-framework-go/pkg/kms"
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+// mock route coordination provider
+type mockProvider struct {
+	openStoreErr error
```
Comment (sandrask) [inline pkg/didcomm/protocol/route/support_test.go]: Error should probably be more generic, are you expecting only openStoreError to be used for testing

File: pkg/didcomm/protocol/route/service_test.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+func TestServiceNew(t *testing.T) {
+	t.Run("test new service - success", func(t *testing.T) {
+		svc, err := New(&mockProvider{})
+		require.NoError(t, err)
+		require.Equal(t, Coordination, svc.Name())
+	})
+
+	t.Run("test new service name - failure", func(t *testing.T) {
+		_, err := New(&mockProvider{openStoreErr: errors.New("error opening the store")})
```
[the remark this answers is not in the document]
Comment (rolsonquadras) [reply pkg/didcomm/protocol/route/service_test.go]: Added as part of a new PR - https://github.com/hyperledger/aries-framework-go/pull/949/files#diff-d8bbd0614f8a1a1ade47173b09ca3e19R32

File: pkg/didcomm/protocol/route/service_test.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	require.Equal(t, true, s.Accept(KeyListUpdateResponseMsgType))
+	require.Equal(t, false, s.Accept("unsupported msg type"))
+}
+
+func TestServiceHandleInbound(t *testing.T) {
+	t.Run("test handle outbound ", func(t *testing.T) {
+		svc, err := New(&mockProvider{})
+		require.NoError(t, err)
+
+		_, err = svc.HandleInbound(nil)
```
[the remark this answers is not in the document]
Comment (rolsonquadras) [reply pkg/didcomm/protocol/route/service_test.go]: This code is removed in new PR - https://github.com/hyperledger/aries-framework-go/pull/949/files#diff-d8bbd0614f8a1a1ade47173b09ca3e19L45

File: pkg/didcomm/protocol/route/support_test.go (hunk 3 above)
Comment (rolsonquadras) [reply pkg/didcomm/protocol/route/support_test.go]: Yes.

**Pull request merged**